src/lib.rs

```diff
@@ -1249,15 +1272,17 @@ impl<T: Counter> Histogram<T> {
     /// covered by previous buckets), and the +1 being used for the lower half of the 0'th bucket.
     /// Or, equivalently, we need 1 more bucket to capture the max value if we consider the
     /// sub-bucket length to be halved.
-    fn num_bins(&self, number_of_buckets: usize) -> usize {
-        (number_of_buckets + 1) * (self.sub_bucket_count / 2)
+    fn num_bins(&self, number_of_buckets: u8) -> u32 {
+        // TODO use sub_bucket_half_count
+        (number_of_buckets as u32 + 1) * (self.sub_bucket_count / 2)
```

Comment:
True. The compiler may well be smart enough to map a integer division by 2 to that anyway, actually... but worth a try if Gil points out some sneaky reason why we want this instead of the field that already has the right answer in it.